interpolation/interpolation.go

```diff
@@ -0,0 +1,116 @@
+// Package interpolation provides methods to interpolate user defined variables and robo tasks.
+package interpolation
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/tj/robo/task"
+	"gopkg.in/yaml.v2"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+var commandPattern = regexp.MustCompile("\\$\\((.+)\\)")
+
+// InterpolateVars interpolates a given map of interfaces (strings or submaps) with itself
+// returning it mit populated template values.
+func InterpolateVars(vars *map[string]interface{}) error {
+	b, err := yaml.Marshal(*vars)
+	if err != nil {
+		return err
+	}
+	s := string(b)
+
+	err = interpolate("variables", *vars, &s)
+	if err != nil {
+		return err
+	}
+
+	err = interpolateVariableCommands(&s)
+	if err != nil {
+		return fmt.Errorf("failed replacing variable placeholder with command result")
+	}
+
+	err = yaml.Unmarshal([]byte(s), vars)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func interpolateVariableCommands(s *string) error {
+	// find all commands
+	matches := commandPattern.FindAllStringSubmatch(*s, -1)
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		cmdOut, err := captureCommandOutput(match[1])
+		if err != nil {
+			return fmt.Errorf("error while executing command. Error: %s", err)
+		}
+		*s = strings.ReplaceAll(*s, match[0], cmdOut)
+	}
+	return nil
+}
+
+// captureCommandOutput executes a command and captures the output which usually gets prompted to stdout.
+func captureCommandOutput(args string) (string, error) {
+	cmd := exec.Command("sh", "-c", args)
```

Comment:
Should we default to the user's active shell?

Reply:
I used `$SHELL` to find a user's active shell and as a fallback it will be still `sh`. Might do the trick, what do you think?

Reply:
sounds great to me :)